server/api/product_handler.py

```diff
@@ -11,21 +11,39 @@
 product_handler = Blueprint('product_handler', __name__)
 
 
-@product_handler.route('/products/<product_id>', methods=['GET', 'DELETE', 'PUT'])
+@product_handler.route('/products/<product_id>', methods=['GET', 'DELETE', 'POST', 'PUT'])
 def oneProductRequests(product_id):
     if request.method == 'GET':
-        product = Product.query.get(int(product_id))
-        return jsonify(product.serialize), 200
+        try:
+            product = Product.query.get(product_id)
+            return jsonify(product.serialize), 200
+        except Exception as e:
+            return jsonify({'error': "{}".format(e.__cause__)}), 400
+
 
     if request.method == 'DELETE':
-        product = Product.query.get(int(product_id))
-        db.session.delete(product)
-        db.session.commit()
-        return jsonify({'response': "Product '{}' was successfully deleted from the database".format(product_id)}), 200
+        try:
+            product = Product.query.get(product_id)
+            db.session.delete(product)
+            db.session.commit()
+            return jsonify({'response': "Product '{}' was successfully deleted from the database".format(product_id)}), 200
+        except Exception as e:
+            return jsonify({'error': "{}".format(e.__cause__)}), 400
     
+    if request.method == 'POST':
+        try:
+            body = json.loads(request.get_data())
+            product_item = Product(product_id, body['name'], body['currency'], body['old_price'], body['price'], body['availability'], body['url'], body['img_url'])
```

Comment:
We need to provide the product_id from the post body as well

Reply:
Wouldn't the product_id be obtained from the route?
`@product_handler.route('/products/<product_id>', methods=['GET', 'DELETE', 'POST', 'PUT'])
`

Reply:
oh right...my bad